pkg/tests/ossm/bug_istiopods_test.go

```diff
@@ -15,79 +15,49 @@
-	bookinfo := examples.Bookinfo{Namespace: "bookinfo"}
-	bookinfo.Install(false)
+		t.LogStep("Install Bookinfo application")
+		app.InstallAndWaitReady(t, app.Bookinfo(ns))
 
-	t.Run("smcp_test_istio_pod_probes_failure", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-		log.Log.Info("Testing: Istio Pod get stuck with probes failure after restart")
-		log.Log.Infof("Create %d new namespaces", numberOfNamespaces)
-		util.Shell(fmt.Sprintf(env.GetRootDir()+"/scripts/smmr/create_members.sh %d", numberOfNamespaces))
-		log.Log.Info("Namespaces created...")
-		rand.Seed(time.Now().UnixNano())
-		// Random number of deletes for the pod between 4 and 10
-		randId := rand.Intn(10-4) + 4
-		log.Log.Info("Random number of deletes: ", randId)
-		count := 0
-		for count < randId {
-			log.Log.Info("*** Get the istiod pod name")
-			istiod, _ := util.GetPodName(`istio-system`, `app=istiod`)
-			log.Log.Info("*** Delete Istio pod: ", istiod)
-			util.Shell(`oc delete pod %s -n istio-system`, istiod)
-			deleted, _ := util.CheckPodDeletion(`istio-system`, `app=istiod`, istiod, 60)
-			if deleted {
-				log.Log.Info("*** Istiod pod deleted: ", istiod)
-				running := util.CheckPodRunning(`istio-system`, `app=istiod`)
-				if running == nil {
-					log.Log.Info("*** New Istiod pod is running")
-					ready, _ := util.CheckPodReady(`istio-system`, `app=istiod`, 10)
-					if ready {
-						log.Log.Info("*** New Istiod pod is ready")
-					} else {
-						istiod, _ = util.GetPodName(`istio-system`, `app=istiod`)
-						// Get events that are not type = Normal from the pod
-						event, _ := util.Shell(`kubectl get events  -n istio-system --field-selector type!=Normal,involvedObject.name=%s |tail -1`, istiod)
-						if strings.Contains(event, "Readiness probe failed") {
-							t.Fatalf("****** Istio pod is not running and fail because of readiness probe failure: %s", event)
-						} else if event == "No resources found in istio-system namespace." {
-							log.Log.Info("*** No eventss for pod: ", istiod)
-						} else {
-							t.Errorf("*** Istio pod is not running but is not failing because of readiness probe failure: %s", event)
-						}
-					}
-					count++
-				}
-			} else {
-				t.Errorf("*** Istiod pod is not deleted: %s", istiod)
-			}
-		}
-		log.Log.Info("*** Test finished without errors after deletion of pods")
+		t.LogStep("Create Namespaces and SMMR")
+		oc.CreateNamespaces(t, multiple_namespaces, data)
+		oc.UpdateSMMRMultipleNamespaces(t, meshNamespace, multiple_smmr, data)
+
+		t.LogStep("Delete Istio pod and check that it is running again")
+		assertIstiodPodReadyAfterDeletion(t, meshNamespace, 10)
 	})
 }
+
+func assertIstiodPodReadyAfterDeletion(t test.TestHelper, ns string, deletionTimes int) {
+	for i := 0; i < deletionTimes; i++ {
+		oc.DeletePod(t, pod.MatchingSelector("app=istiod", ns))
+		oc.WaitPodRunning(t, pod.MatchingSelector("app=istiod", ns))
+		oc.WaitPodReady(t, pod.MatchingSelector("app=istiod", ns))
+	}
+}
```

Comment:
What's the benefit of having this function instead of having its code inlined? I think the function just hides things. When I read the test, I always have to go look at the function to see what exactly it's doing (the name of the function doesn't say that the function restarts istiod multiple times).

Reply:
Also, this just calls `WaitPodReady`. If the pod fails to become ready, this test will fail with a weird timeout error instead of clearly showing what's wrong. I would expect this test to print something like: 
```
ERROR: istiod pod failed to become ready after restart
```

Reply:
Added to oc.WaitPodReady:
```go
	condition := shell.Executef(t, "kubectl -n %s wait --for condition=Ready pod %s --timeout 30s", pod.Namespace, pod.Name)
	if strings.Contains(condition, "condition met") {
		t.Logf("Pod %s in namespace %s is ready!", pod.Name, pod.Namespace)
	} else {
		t.Fatalf("Error: %s in namespace %s is not ready: %s", pod.Name, pod.Namespace, condition)
	}
```
To validate if the condition is met or not and give to the user a more clear message